### ISSUE TITLE
Switch VGA text mode

### ISSUE THREAD

**Issue by FrankRay78** [FrankRay78/PatienceOS#11]
See [VGA text mode](https://en.wikipedia.org/wiki/VGA_text_mode)

**Comment by FrankRay78** [comment]
> I used this code in my very old project to change from 80x25 to 80x50,

```asm
; set 80x50 text mode
   mov   ax,1112h   ; AH=11h TEXT-MODE CHARACTER GENERATOR FUNCTIONS
            ; AL=12h load ROM 8x8 double-dot pattern (PS,EGA,VGA)
   xor   bl,bl      ; block to load
   int   10h
```

https://forum.osdev.org/viewtopic.php?p=334528#p334528

> Mode 13h is the standard 256-color mode on VGA graphics hardware introduced in 1987 with the IBM PS/2. It has a resolution of 320×200 pixels.

https://en.wikipedia.org/wiki/Mode_13h

> Can someone please send me a working code for setting up VGA mode 13?

```asm
    mov ax,0x0013
    int 0x10
```

> It assumes that the OS is designed properly and therefore sets a video mode in real mode during boot
> It assumes that the computer was booted using BIOS (and not UEFI)

And also

> [The first two problems] can be mitigated a little by using ["int 0x10, ax = 0x1A00"](http://www.ctyme.com/intr/rb-0219.htm) to check that a video card that emulates VGA at the video ROM level exists.

https://forum.osdev.org/viewtopic.php?p=280800#p280800

**Comment by FrankRay78** [comment]
An extensive listing of all the various INT10, video function calls:

![image](https://github.com/FrankRay78/PatienceOS/assets/52075808/722abec4-6f15-4f61-bcd5-aec064d2be23)

https://www.ctyme.com/intr/int-10.htm

**Comment by FrankRay78** [comment]
Assembly method to print a string to the console: https://github.com/cfenollosa/os-tutorial/blob/ce8e050d246594fda7f0067d7f4c9abeaa2b269c/15-video-ports/boot/print.asm

**Comment by FrankRay78** [comment]
TLDR: The PatienceOS kernel is booting in 32-bit protected mode, by default. 

The following code, when added into `loader.asm` has no effect:

```asm
        mov ah, 0x00    ; Video function: Set Video Mode
        mov al, 0x00    ; Video mode: 00h (T40x25)
        int 0x10        ; Call BIOS video interrupt
```

After hours of trying to work out why, I notice the following comment on the [OS Dev: Bare Bones tutorial](https://wiki.osdev.org/Bare_Bones):

```asm
_start:
	/*
	The bootloader has loaded us into 32-bit protected mode on a x86
	machine. Interrupts are disabled
```

I believe this explains why I can't seem to make a call via int 10 within `_start:`

**Comment by FrankRay78** [comment]
The following works nicely, I've tested it:

```asm
mov ax, 0x0003 ; VGA text mode, 720x400 pixels, 9x16 font, 80x25 characters
int 0x10
mov ax, 0x1112 ; replace 9x16 font with 9x8 font, so now it's 80x50 characters
mov bl, 0x00
int 0x10
```
